bridges/ethMultiversX/bridgeExecutor.go

```diff
@@ -169,6 +185,47 @@ func (executor *bridgeExecutor) StoreBatchFromMultiversX(batch *bridgeCore.Trans
 	return nil
 }
 
+// addBatchSCMetadataMvx fetches the logs containing sc calls metadata for the current batch
+func (executor *bridgeExecutor) addBatchSCMetadataMvx(ctx context.Context, transfers *bridgeCore.TransferBatch) (*bridgeCore.TransferBatch, error) {
+	if transfers == nil {
+		return nil, ErrNilBatch
+	}
+
+	events, err := executor.multiversXClient.GetBatchSCMetadata(ctx, transfers)
+	if err != nil {
+		return nil, err
+	}
+
+	for i, t := range transfers.Deposits {
+		transfers.Deposits[i], err = executor.addMetadataToTransferMvx(t, events)
```

Comment:
no need to return the transfer from addMetadataToTransferMvx and reassign it, as the method gets a pointer to it as parameter

similar to processData

Reply:
still not implemented